sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuOverrides.scala

```diff
@@ -2284,13 +2284,16 @@ object GpuOverrides {
       }),
     expr[CreateArray](
       " Returns an array with the given elements",
+      // Cudf only supports 2D Array.
```

Comment:
This is not true cudf can support very nested arrays.  If a nested type is in nested, then it implies recursion.

```
TypeSig.Array.nested(Typeside.BOOLEAN + TypeSig.ARRAY)
```
Matches `ArrayType(BooleanType)` and `ArrayType(ArrayType(BooleanType))` but also as deeply nested as we want to go. `ArrayType(ArrayType(ArrayType(...(BooleanType))))))))))))))))`

If you only want one level deep then we re going to have to make some changes to how the type checking code works.

Reply:
@revans2, Yeah, For now, seems the API makeList can only create 2D Array. So Seems the current TypeCheck is no longer cover this scenario.  Can you share some ideas about this?